docs/source/conf.py

```diff
@@ -51,7 +50,6 @@
 # Configure intersphinx magic
 intersphinx_mapping = {
     'twisted': ('https://twistedmatrix.com/documents/current/api/', None),
-    'pydoctor': ('https://pydoctor.readthedocs.io/en/latest/api/', None),
```

Comment:
Note that this URL was wrong...as when building on Read The docs for the "stable" version, you will still have links to "latest"

With this PR, this should be fixed. 

Once we have a new release, we should check that we have separate API links for

See `You can subclass the pydoctor.zopeinterface.ZopeInterfaceSystem and pass your custom class `

stable vs  latest

* https://pydoctor.readthedocs.io/en/latest/usage.html#use-a-custom-system-class
* https://pydoctor.readthedocs.io/en/stable/usage.html#use-a-custom-system-class

Right now the API links is always latest.